asdf/asdf.py

```diff
@@ -344,7 +376,8 @@ def _check_extensions(self, tree, strict=False):
                 if Version(installed.package_version) < Version(extension.software["version"]):
                     msg = (
                         f"File {filename}was created with extension {extension_description}, "
-                        f"but older package ({installed.package_name}=={installed.package_version}) is installed."
+                        f"but older package ({installed.package_name}=={installed.package_version}) is installed. "
+                        "This warning will be removed in 3.0."
```

Comment:
Can we keep this warning and just not force it to be turned into an error? We could add another `AsdfWarning` subclass (perhaps `AsdfExtensionVersionWarning`) and link to the documentation on how to convert these warnings into errors.